zcm/julia/ZCM.jl

```diff
@@ -0,0 +1,334 @@
+end
+export stop;
+
+function handle(zcm::Zcm)
+    return ccall(("zcm_handle", "libzcm"), Cint, (Ptr{Native.Zcm},), zcm.zcm);
+end
+export handle;
+
+function handle_nonblock(zcm::Zcm)
+    return ccall(("zcm_handle_nonblock", "libzcm"), Cint,
+                 (Ptr{Native.Zcm},), zcm.zcm);
+end
+export handle_nonblock;
+
+# RRR: go over the signedness of all these types from within zcm ... some of them are dumb
+type LogEvent
+    event   ::Ptr{Native.EventLogEvent};
+    num     ::Int64;
+    utime   ::Int64;
+    channel ::String;
+    data    ::Array{UInt8};
+
+    function LogEvent(event::Ptr{Native.EventLogEvent})
+        instance  = new();
+        instance.event = event;
+
+        if (event != C_NULL)
+            loadedEvent = unsafe_load(event);
+
+            instance.num   = loadedEvent.eventnum;
+            instance.utime = loadedEvent.timestamp;
+            if (loadedEvent.channel != C_NULL)
+                instance.channel = unsafe_string(loadedEvent.channel, loadedEvent.chanlen);
+            else
+                instance.channel = ""
+            end
+            if (loadedEvent.data != C_NULL)
+                instance.data    = unsafe_wrap(Array, loadedEvent.data, loadedEvent.datalen);
+            else
+                instance.data    = [];
+            end
+        end
+
+        # user can force cleanup of their instance by calling `finalize(zcm)`
+        finalizer(instance, function(event::LogEvent)
+                                if (event.event != C_NULL)
+                                    ccall(("zcm_eventlog_free_event", "libzcm"), Void,
+                                          (Ptr{Native.EventLogEvent},), event.event);
+                                    event.event = C_NULL;
+                                end
+                            end);
+
+        return instance;
+    end
+end
+export LogEvent;
+
+type LogFile
+    eventLog::Ptr{Native.EventLog};
+
+    function LogFile(path::AbstractString, mode::AbstractString)
+        println("Creating zcm eventlog from path : ", path, " with mode ", mode);
+        instance = new();
+        instance.eventLog = ccall(("zcm_eventlog_create", "libzcm"), Ptr{Native.EventLog},
+                                  (Cstring, Cstring), path, mode);
+
+        # user can force cleanup of their instance by calling `finalize(zcm)`
+        finalizer(instance, function(log::LogFile)
+                                if (log.eventLog != C_NULL)
+                                    ccall(("zcm_eventlog_destroy", "libzcm"), Void,
+                                          (Ptr{Native.EventLog},), log.eventLog);
+                                    log.eventLog = C_NULL;
+                                end
+                            end);
+
+        return instance;
+    end
+end
+export LogFile;
+
+function good(lf::LogFile)
+    return lf.eventLog != C_NULL;
+end
+export good;
+
+function readNextEvent(lf::LogFile)
```

Comment:
Lower case and underscored is standard in Julia, but again I understand if you want to make things have the same name between languages.